src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -194,46 +194,55 @@ std::vector<size_t> LinearIR::LoopManager::get_outer_expr_loops(const Expression
     return std::vector<size_t>(loop_ids.cbegin(), it);
 }
 
-void LinearIR::LoopManager::get_loop_bounds(const LinearIR &linear_ir,
-                                            size_t loop_id,
-                                            LinearIR::constExprIt &loop_begin_pos,
-                                            LinearIR::constExprIt &loop_end_pos,
-                                            bool loop_ops_inserted) const {
+std::pair<LinearIR::constExprIt, LinearIR::constExprIt> LinearIR::LoopManager::get_loop_bounds(const LinearIR &linear_ir, size_t loop_id) const {
     const auto loop_info = get_loop_info(loop_id);
-    get_loop_bounds(linear_ir, loop_info->get_entry_points(), loop_info->get_exit_points(), loop_begin_pos, loop_end_pos, loop_id, loop_ops_inserted);
-}
-
-void LinearIR::LoopManager::get_loop_bounds(const LinearIR &linear_ir,
-                                            const std::vector<LoopPort>& entries,
-                                            const std::vector<LoopPort>& exits,
-                                            LinearIR::constExprIt &loop_begin_pos,
-                                            LinearIR::constExprIt &loop_end_pos,
-                                            size_t loop_id, bool loop_ops_inserted) {
-    OPENVINO_ASSERT(!entries.empty(), "Loop must have entry points");
-    OPENVINO_ASSERT(!exits.empty(), "Loop must have entry points");
-    const auto& entry_expr = entries.front().expr_port->get_expr();
-    loop_begin_pos = linear_ir.find(entry_expr);
-
-    // Some operations in Loop can be before first entry points: Scalars, VectorBuffer.
-    // We should iterate by them till the expr is in the corresponding Loop
-    auto prev_loop_ids = (*std::prev(loop_begin_pos))->get_loop_ids();
-    while (std::find(prev_loop_ids.begin(), prev_loop_ids.end(), loop_id) != prev_loop_ids.end()) {
-        loop_begin_pos = std::prev(loop_begin_pos);
-        prev_loop_ids = (*std::prev(loop_begin_pos))->get_loop_ids();
-    }
+    return get_loop_bounds(linear_ir, loop_id, loop_info->get_entry_points(), loop_info->get_exit_points());
+}
+
+std::pair<LinearIR::constExprIt, LinearIR::constExprIt> LinearIR::LoopManager::get_loop_bounds(const LinearIR &linear_ir, size_t loop_id,
+                                                                                               const std::vector<LoopPort>& entries,
+                                                                                               const std::vector<LoopPort>& exits) {
+    auto is_explicit_loop = [&loop_id](const ExpressionPtr& expr) {
+        if (const auto loop_end = ov::as_type_ptr<op::LoopEnd>(expr->get_node()))
+            return loop_end->get_id() == loop_id;
+        return false;
+    };
+    // There no explicit Loop expressions
+    if (std::none_of(linear_ir.cbegin(), linear_ir.cend(), is_explicit_loop)) {
```

Comment:
Why the logic for inserted and not-inserted explicit `Loop*` ops differs so drastically?
For example, `LoopBegin` for a given loop can't be in anywhere in the IR, right? It must be before the first entry point (and possibly some extra Scalars/VectorBuffers). Then why do we need to walk through the whole IR on L211, and then walk through the whole IR again on L234?
In other words, why can't we reuse the logic on L212-229 for both cases?

Reply:
Discussed offline: the main problem was that `LoopEnd` has `PortConnectors` that aren't mapped on `LoopInfo`. For that I've created the PR https://github.com/openvinotoolkit/openvino/pull/22161. With this feature we can align case without existing loop expressions and with them to get loop bounds.